op-validator/README.md

```diff
@@ -0,0 +1,47 @@
+# op-validator
+
+The op-validator is a tool for validating Optimism chain configurations and deployments. It works by calling into the StandardValidator smart contracts (StandardValidatorV180 and StandardValidatorV200). These then perform a set of checks, and return error codes for any issues found. These checks include:
+
+- Contract implementations and versions
+- Proxy configurations
+- System parameters
+- Cross-component relationships
+- Security settings
+
+## Usage
+
+The validator supports different protocol versions through subcommands:
+
+```bash
+op-validator validate [version] [flags]
+```
+
+Where version is one of:
+
+- `v1.8.0` - For validating protocol version 1.8.0
+- `v2.0.0` - For validating protocol version 2.0.0
+
+### Required Flags
+
+- `--l1-rpc-url`: L1 RPC URL (can also be set via L1_RPC_URL environment variable)
+- `--absolute-prestate`: Absolute prestate as hex string
```

Comment:
(non-blocking comment) Is the pre-state hex string just a serialized/hashed version of an originally raw JSON representation containing all the information about the pre-state.

Or is it just the state-root / block hash of the last block before the provided protocol version?

"Assuming I understand 'pre-state' here correctly, which is, the state of the chain just before a point in time, in our case, which is the protocol version?"

Reply:
> (non-blocking comment) Is the pre-state hex string just a serialized/hashed version of an originally raw JSON representation containing all the information about the pre-state.

It commits to the contents of the `op-program`, which is a JSON that contains a bunch of info (symbols/instructions/etc.) for the program.